File: Owner_module_backend/src/main/java/com/backend/ownerentity/Owner.java
package com.backend.ownerentity;

import com.backend.userentity.User;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "owners")
@Inheritance(strategy = InheritanceType.JOINED)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@com.fasterxml.jackson.annotation.JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Owner {

    @Id
    @Column(name = "owner_id")
    private Long ownerId;

    // Shared primary key with User
    @OneToOne
    @MapsId
    @JoinColumn(name = "owner_id")
    @JsonIgnore
    private User user;

    @Convert(converter = OwnerTypeConverter.class)
    @Column(name = "owner_type", nullable = false)
    private OwnerType ownerType;   // PG / MESS

    @Column(name = "name", nullable = false, length = 100)
    private String name;

    @Column(name = "contact_no", nullable = false, length = 15)
    private String contactNo;

    @Column(name = "email", nullable = false, unique = true)
    private String email;

    @Column(name = "address", nullable = false)
    private String address;

    @Convert(converter = OwnerStatusConverter.class)
    @Column(name = "status", nullable = false)
    private OwnerStatus status;  // ACTIVE / INACTIVE / PENDING

    @Convert(converter = IdCardTypeConverter.class)
    @Column(name = "id_card_type", nullable = false)
    private IdCardType idCardType;

    @Column(name = "id_card_number", nullable = false, unique = true)
    private String idCardNumber;
}

File: AdminService/Entities/Owner.cs
//using System.ComponentModel.DataAnnotations.Schema;

//namespace AdminService.Entities
//{
//    [Table("owners")]
//    public class Owner
//    {
//        public long OwnerId { get; set; }

//        [Column("owner_type")]
//        public string OwnerType { get; set; }

//        [Column("status")]

//        public string Status { get; set; }

//        public string Name { get; set; }

//        public string ContactNo { get; set; }

//        public string Address { get; set; }

//        public string IdCardType { get; set; }

//        public string IdCardNumber { get; set; }

//        public long UserId { get; set; }
//    }
//}



using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdminService.Entities
{
    [Table("owners")]
    public class Owner
    {
        [Key]
        [Column("owner_id")] // Matches @Column(name = "owner_id")
        public long OwnerId { get; set; }

        [Column("owner_type")] // Matches @Column(name = "owner_type")
        public string OwnerType { get; set; } // Enums map to strings in your Java setup

        [Column("name")]
        public string Name { get; set; }

        [Column("contact_no")]
        public string ContactNo { get; set; }

        [Column("email")]
        public string Email { get; set; }

        [Column("address")]
        public string Address { get; set; }

        [Column("status")]
        public string Status { get; set; }

        [Column("id_card_type")]
        public string IdCardType { get; set; }

        [Column("id_card_number")]
        public string IdCardNumber { get; set; }

        // Navigation property (Optional, but useful for Joins)
        [ForeignKey("OwnerId")]
        public virtual User User { get; set; }
    }
}